pkg/cmd/server/server.go

```diff
@@ -274,6 +285,13 @@ func (s *server) run() error {
 		return errors.WithStack(err)
 	}
 
+	for _, vslName := range s.config.defaultVolumeSnapshotLocations {
```

Comment:
We also want to validate that every provider has a default VSL specified.  So you probably want to `s.ArkClient.ArkV1().VolumeSnapshotLocations(s.namespace).List(...)`, iterate through it to get a unique list of providers (`volumeSnapshotLocation.Spec.Provider`), then make sure that each of them had a value provided in the map, and that the value corresponds to a VSL **for that provider**.